jobs/AnalyzeResults.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace MediaSearchSignalTest\Jobs;
+
+use mysqli;
+
+class AnalyzeResults {
+
+    private $db;
+    private $searchId;
+    private $out;
+
+    public function __construct( array $config, int $searchId ) {
+        $this->db = new mysqli( $config['db']['host'], $config['client']['user'],
+            $config['client']['password'], $config['db']['dbname'] );
+        if ( $this->db->connect_error ) {
+            die('DB connection Error (' . $this->db->connect_errno . ') '
+                . $this->db->connect_error);
+        }
+        $this->out = fopen(
+            __DIR__ . '/../out/AnalyzeResults_' . $searchId . '.csv',
+            'w'
+        );
+        $this->searchId = $searchId;
+    }
+
+    public function __destruct() {
+        fclose( $this->out );
+    }
+
+    public function run() {
+        $resultsets = $this->db->query(
+            'select id,term from resultset where searchId = ' . intval( $this->searchId )
+        )->fetch_all( MYSQLI_ASSOC );
+        if ( count( $resultsets ) == 0 ) {
+            die( "ERROR: no results found with for search id " . $this->searchId . "\n");
+        }
+
+        fwrite(
+            $this->out,
+            "Term,f1Score\n"
+        );
+
+        $f1Scores = $precisionTop30s = [];
+        foreach ( $resultsets as $resultset ) {
+            $f1Scores[$resultset['term']] = $this->calculateF1Score(
+                $resultset['id'], $resultset['term'] );
+            $precisionTop30s[$resultset['term']] = $this->calculatePrecisionTop30(
+                $resultset['id'], $resultset['term'] );
+            fwrite( $this->out,
+                $resultset['term'] . "," .
+                $f1Scores[$resultset['term']] . "," .
+                $precisionTop30s[$resultset['term']] . "\n" );
+        }
+        fwrite(
+            $this->out,
+            "ARITHMETIC MEAN," .
+            array_sum( $f1Scores ) / count( $resultsets) . "," .
+            array_sum( $precisionTop30s ) / count( $resultsets) .
+            "\n"
+        );
+    }
+
+    private function calculateF1Score( int $resultsetId, string $searchTerm ) : float {
+        $knownGoodImageCount = $this->db->query(
```

Comment:
When knownGoodImageCount === 0, we should completely ignore this term (if there is no known good data, then the algorithm can't find it, and we can't assess how well it does)